pkg/metrics/collector/sockstat.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type sockStatCollector struct {
+	inuse  *prometheus.Desc
+	tw     *prometheus.Desc
+	memory *prometheus.Desc
+}
+
+var pageSize = os.Getpagesize()
```

Comment:
why cache this value?

Reply:
Because the number we read from the proc file is the number of pages. To get the actual memory usage, we need to multiply the number to the page size.